modules/gdscript/editor/gdscript_highlighter.cpp

```diff
@@ -432,6 +433,11 @@ Dictionary GDScriptSyntaxHighlighter::_get_line_syntax_highlighting_impl(int p_l
 				col = member_keywords[word];
 			}
 
+			// Don't highlight for loop inline bodies as types.
+			if (expect_type && in_for_loop && !class_names.has(word)) {
```

Comment:
This is incorrect, types may not only be registered classes. The highlighter should recognize patterns like:

```
"for" WS+ Identifier WS* ":" WS* Type

where:

WS ::= " " | "\t"
Identifier ::= ( ID_Start | "_" ) ID_Continue*
Type ::= IdentifierChain ( "[" Type ( "," Type )* "]" )?
IdentifierChain ::= Identifier ( "." Identifier )* 
```